src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5308,7 +5324,10 @@ private VisitArgumentResult VisitArgumentEvaluate(BoundExpression argument, RefK
                             stateForLambda: result.StateForLambda);
 
                         // If the parameter has annotations, we perform an additional check for nullable value types
```

Comment:
>nullable value types [](start = 96, length = 20)

Is this just for nullable value types?

Reply:
Yes, for example in the case of a parameter `[DisallowNull] string? s` we can adjust the type to just `string` and then visit the conversion. But in the case of `[DisallowNull] int? i` we can't adjust the type in that way. [SharpLab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+BiAdgVwBtCJhC4ACOXU8gWACgABAJgEZGmAGCptgOgAiASwgBzXAHtYwgMZR+AYUko4AQRqEAnlGFQA3I04BmXiwqKKAb0YU7vACwUAsmwAUASlv2bDe/5cAZRgEYVwxNwJiD0M/ALtnAElcGEiiQhjvOwBfLIo8plMmJ2dg0PC3AG0RKAhiSQB3ADl0gF1eNi4AfgooD2s83Li7AqKS5NTqvTrCRpbidrCYHuF+338h7KA). Note particularly the difference in diagnostic IDs.